### PR TITLE
Update azuresearch.py following recent change from azure-search-documents library

### PR DESCRIPTION
**Pull request by guillaumedelande**
- **Description:** 

Reference library azure-search-documents has been adapted in version 11.4.0:

1. Notebook explaining Azure AI Search updated with most recent info
2. HnswVectorSearchAlgorithmConfiguration --> HnswAlgorithmConfiguration
3. PrioritizedFields(prioritized_content_fields) --> SemanticPrioritizedFields(content_fields)
4. SemanticSettings --> SemanticSearch
5. VectorSearch(algorithm_configurations) --> VectorSearch(configurations)

--> Changes now reflected on Langchain: default vector search config from langchain is now compatible with officially released library from Azure.

  - **Issue:**
Issue creating a new index (due to wrong class used for default vector search configuration) if using latest version of azure-search-documents with current langchain version
  - **Dependencies:** azure-search-documents>=11.4.0,
  - **Tag maintainer:** ,

### CODE REVIEW

File: libs/langchain/langchain/vectorstores/azuresearch.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def _get_search_client(
     from azure.search.documents import SearchClient
     from azure.search.documents.indexes import SearchIndexClient
     from azure.search.documents.indexes.models import (
-        HnswVectorSearchAlgorithmConfiguration,
+        HnswAlgorithmConfiguration, # HnswVectorSearchAlgorithmConfiguration outdated in version 11.4.0
```
Comment (efriis) [inline libs/langchain/langchain/vectorstores/azuresearch.py:84]: Can we do both in a try/except for people on an old sdk version?

**Review by efriis — COMMENTED** [review]
Thanks! 1 small change for folks still using the old sdk

**Comment by guillaumedelande** [comment]
Have added some other changes

**Comment by baskaryan** [comment]
cc @ruoccofabrizio

**Review by efriis — APPROVED** [review]

**Pull request merged**